src/components/ui/button/Button.vue

```diff
@@ -68,6 +68,13 @@ if (props.disclosure) {
 	trailingIcon.value = 'arrow_drop_down'
 	classList.value.push('-disclousure')
 }
+
+watch(
+	() => props.variant,
+	(newVal) => {
+		classList.value.push(`-${newVal}`)
+	}
```

Comment:
Aqui não seria interessante remover da classList a variant antiga, após fazer o push, para não fica acumulando as classes de variant no botão?
```suggestion
	(newVal, oldVal) => {
		classList.value.push(`-${newVal}`)
		const index = classList.value.indexOf(`-${oldVal}`);
                if (index !== -1) {
                  classList.value.splice(index, 1);
                }
		               
	}
```

Reply:
pior que fiz esse teste aqui e esse array de class list é remontado a cada atualização de prop mano

Reply:
Boa mano, então vou aprovar aqui ;)